endpoints/event.go

```diff
@@ -0,0 +1,367 @@
+	IMP Type = "imp"
+
+	BLANK Format = "b"
+	IMAGE Format = "i"
+
+	ENABLED  Analytics = "1"
+	DISABLED Analytics = "0"
+
+	// Required
+	TEMPLATE_URL         = "%v/event?t=%v&b=%v&a=%v"
+	TYPE_PARAMETER       = "t"
+	BID_ID_PARAMETER     = "b"
+	ACCOUNT_ID_PARAMETER = "a"
+
+	// Optional
+	BIDDER_PARAMETER    = "bidder"
+	TIMESTAMP_PARAMETER = "ts"
+	FORMAT_PARAMETER    = "f"
+	ANALYTICS_PARAMETER = "x"
+
+	TRACKING_PIXEL_PNG              = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAABHNCSVQICAgIfAhkiAAAAA1JREFUCJljYGBgYAAAAAUAAYehTtQAAAAASUVORK5CYII="
+	TRACKING_PIXEL_PNG_CONTENT_TYPE = "image/png"
+)
+
+type EventRequest struct {
+	Type      Type      `json:"type,omitempty"`
+	Bidid     string    `json:"bidid,omitempty"`
+	AccountId string    `json:"account_id,omitempty"`
+	Bidder    string    `json:"bidder,omitempty"`
+	Timestamp int64     `json:"timestamp,omitempty"`
+	Format    Format    `json:"format,omitempty"`
+	Analytics Analytics `json:"analytics,omitempty"`
+}
+
+type TrackingPixel struct {
+	Content     []byte `json:"content,omitempty"`
+	ContentType string `json:"content_type,omitempty"`
+}
+
+type eventEndpoint struct {
+	DataCache     cache.Cache
+	Analytics     analytics.PBSAnalyticsModule
+	TrackingPixel *TrackingPixel
+}
+
+func NewEventEndpoint(dataCache cache.Cache, analytics analytics.PBSAnalyticsModule) httprouter.Handle {
+	ee := &eventEndpoint{
+		DataCache:     dataCache,
+		Analytics:     analytics,
+		TrackingPixel: loadTrackingPixel(),
+	}
+
+	return ee.Handle
+}
+
+func (e *eventEndpoint) Handle(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	// parse event request from http req
+	eventRequest, err := ParseEventRequest(r)
+
+	// handle possible parsing errors
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("invalid request: %s\n", err.Error())))
+
+		return
+	}
+
+	// validate account id
+	accountId, err := validateRequiredParameter(r, ACCOUNT_ID_PARAMETER)
+
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(fmt.Sprintf("Account '%s' is required query parameter and can't be empty", ACCOUNT_ID_PARAMETER)))
+		return
+	}
+	eventRequest.AccountId = accountId
+
+	if eventRequest.Analytics == ENABLED {
+
+		// get account details
+		account, err := e.DataCache.Accounts().Get(eventRequest.AccountId)
+		if err != nil {
+			if err == sql.ErrNoRows {
+				account = &cache.Account{
+					ID:            accountId,
+					EventsEnabled: false,
+				}
+			} else {
+				if glog.V(2) {
+					glog.Infof("Invalid account id: %v", err)
+				}
+
+				status := http.StatusInternalServerError
+				message := fmt.Sprintf("Invalid request: %s\n", err.Error())
+
+				w.WriteHeader(status)
+				w.Write([]byte(message))
+				return
+			}
+		}
+
+		// account does not support events
+		if !account.EventsEnabled {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte(fmt.Sprintf("Account '%s' doesn't support events", eventRequest.AccountId)))
```

Comment:
Are these the same responses given by the PBS-Java endpoint? It would be good to match what they already have in place.

Reply:
Yes, it is: https://github.com/rubicon-project/prebid-server-java/blob/master/src/main/java/org/prebid/server/handler/NotificationEventHandler.java#L132